### PR TITLE
Added deletion of users' PMs in usernuke plugin

### PR DESCRIPTION
**Pull request by AlphaT3ch**
Added the deletion of the nuked users' PMs. This is in-case they used a PM spam script on any user of the site. This is in effect as plugins/usernuke/page_nuke.php. This affects the usernuke plugin.

### CODE REVIEW

File: plugins/usernuke/page_nuke.php [hunk 1]
```diff
@@ -44,6 +44,14 @@
...
 		query("delete from {usercomments}
 				where uid={0} or cid={0}", $uid);
+		//Delete all of their private messages
+		$delID = query("select id from {pmsgs} where userfrom={0}", $uid);
+		if ($result->num_rows > 0) {
+			while($delID = $toDelID->fetch_assoc()) {
+			query("delete from {pmsgs_text} where pid={0}", $todelID);
+			}
+		} else {}
+		query("delete from {pmsgs} where userfrom={0}", $uid);
```
Comment (StapleButter) [inline plugins/usernuke/page_nuke.php]: man what the hell

how is that code even supposed to work

**Comment by Dirbaio** [comment]
Oh, wow, I can't believe I forgot about PM's. Thanks for the PR!

Could you change the pmsgs_text deletion to be a single query? Looping and deleting row by row is quite inefficient.

Something like this should work:

```SQL
DELETE pt FROM pmsgs_text
LEFT JOIN pmsgs p ON pt.pid = p.id
WHERE p.userfrom = {0}
```

**Comment by AlphaT3ch** [comment]
@StapleButter It works because my code it amazing! (and that I looked at the one at w3schools)
@Dirbaio I've added the code and I think I've put it where it needs to be to work.

**Comment by AlphaT3ch** [comment]
So I just finally got to test that code. The code that I originally provided with the foreach seemed to work perfectly, but would be very inefficient on large boards. The current one with the code from @Dirbaio brins up a 503 vanish cache error on my server. My server does randomly throw 503 often though. I'm not quite sure at the moment.

**Comment by LifeMushroom** [comment]
Is this gonna be merged, because I keep seeing boards that have been fucked up because of no PM deletion <_>

**Comment by AlphaT3ch** [comment]
@LifeMushroom: I'm wondering that myself... @Dirbaio

**Comment by LifeMushroom** [comment]
Actually, @Dirbaio @AlphaT3ch 

query("DELETE pt FROM pmsgs_text LEFT JOIN pmsgs p ON pt.pid = p.id WHERE p.userfrom = {0}", $uid);

That made the whole thing spout out a SQL error. It didn't nuke. When I removed that line and kept
query("delete from {pmsgs} where userfrom={0}", $uid);

and nuked them, it worked and deleted all their PM's.

**Pull request closed without merging**